### DISCUSSION TITLE
Deploy Kelemetry for Production Clusters failed at the helm installation

### DISCUSSION THREAD

**Discussion by calvinxu** [kubewharf/kelemetry#120] [Q&A] [answered]
# helm install kelemetry oci://ghcr.io/kubewharf/kelemetry-chart --values values.yaml --debug
install.go:200: [debug] Original chart version: ""
Error: INSTALLATION FAILED: GET "https://ghcr.io/v2/kubewharf/kelemetry-chart/tags/list": unable to retrieve credentials
helm.go:84: [debug] GET "https://ghcr.io/v2/kubewharf/kelemetry-chart/tags/list": unable to retrieve credentials
INSTALLATION FAILED
main.newInstallCmd.func2
        helm.sh/helm/v3/cmd/helm/install.go:147
github.com/spf13/cobra.(*Command).execute
        github.com/spf13/cobra@v1.6.1/command.go:916
github.com/spf13/cobra.(*Command).ExecuteC
        github.com/spf13/cobra@v1.6.1/command.go:1044
github.com/spf13/cobra.(*Command).Execute
        github.com/spf13/cobra@v1.6.1/command.go:968
main.main
        helm.sh/helm/v3/cmd/helm/helm.go:83
runtime.main
        runtime/proc.go:250
runtime.goexit
        runtime/asm_amd64.s:1598

**Comment by calvinxu** [comment]
also, the step
"If you use an audit webhook directly, remember to [configure the apiserver](https://kubernetes.io/docs/tasks/debug/debug-cluster/audit/#webhook-backend) to send audit logs to the webhook", according to the comment in values.yaml
"# `webhook` runs an audit webhook server together with the consumer,
    # and directly passes the events to the consumer in an in-memory channel.
    # This is the simplest setup if you run your own K8S cluster.
    # Manual configuration of kube-apiserver audit webhook is still REQUIRED;
    # see <https://kubernetes.io/docs/tasks/debug/debug-cluster/audit/#webhook-backend> for details.
"
does it mean the webhook server start up by kelemetry? and what we do for configuration webhook is to configure the kube-apiserver to enable the audit-log with webhook setting? what is the webhook server url if it starts by kelemetry?

In the quickstart demo, I saw the webhook config like this:
```
apiVersion: v1
clusters:
- cluster:
    server: http://172.21.0.1:8080/audit/tracetest
  name: audit
contexts:
- context:
    cluster: audit
  name: main
current-context: main
kind: Config
preferences: {}
```

how can I know that "server: http://172.21.0.1:8080/audit/tracetest" that kelemetry runs similiar as quickstart demo?

**Comment by SOF3** [comment, marked as answer]
Yes, if you enable the webhook option, the webhook server is deployed together in the consumer instances. If you have an ingress controller configured, you can try `curl -I -X POST /audit`; a `200 OK` response shows that the audit server is running.

The part behind `/audit/` is supposed to your cluster name, as specified in values.yaml `.multiCluster.clusters[].name`.
